pre_commit/clientlib.py

```diff
@@ -155,6 +156,9 @@ def ordered_load_normalize_legacy_config(contents):
         # TODO: Once happy, issue a deprecation warning and instructions
         data = collections.OrderedDict([('repos', data)])
 
+    for repo_config in data['repos']:
+        repo_config['repo'] = os.path.expanduser(repo_config['repo'])
+
```

Comment:
my other hesitation is I plan to *delete* this function after the deprecation period.